### PR TITLE
Single Env Filters with Pagination 

### PR DESCRIPTION
**Pull request by kotwani2883**
## Explain the feature/fix

Adding filters in tables on total SPAship 

[screen-capture (5).webm](https://user-images.githubusercontent.com/56580582/211316852-9ff00b77-5672-4290-84cb-9d438f28a124.webm)

### CODE REVIEW

File: packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx [hunk 1]
```diff
@@ -58,27 +62,32 @@ const URL_LENGTH_LIMIT = 50;
 export const WebPropertyDetailPage = (): JSX.Element => {
   const { query } = useRouter();
   const [isRowExpanded, setIsRowExpanded] = useState<Record<string, boolean>>({});
+  const [searchTerm, setSearchTerm] = useState('');
+  const [filtervalue, setFilter] = useState('');
```
Comment (akhilmhdh) [inline packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx:66]: change this to a better name like filterByEnv and setFilterByEnv as this name is too generic

File: packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx [hunk 2]
```diff
@@ -58,27 +62,32 @@ const URL_LENGTH_LIMIT = 50;
...
   // api calls
-  const spaProperties = useGetSPAPropGroupByName(propertyIdentifier);
+  const  spaProperties = useGetSPAPropGroupByName(propertyIdentifier,filtervalue);
   const webProperties = useGetWebPropertyGroupedByEnv(propertyIdentifier);
   const ephemeralPreview = useGetEphemeralListForProperty(propertyIdentifier);
-
+  const spaPropertyKeys = Object.keys(spaProperties.data || {});
+  const isSpaPropertyListEmpty = spaPropertyKeys.length === 0;
+  const webPropertiesKeys = Object.keys(webProperties.data || {});
+  const iswebPropertiesListEmpty = webPropertiesKeys.length === 0;
```
Comment (akhilmhdh) [inline packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx:81]: camelCase Please

File: packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx [hunk 3]
```diff
@@ -131,12 +143,48 @@ export const WebPropertyDetailPage = (): JSX.Element => {
...
+              }}
+              
+              // selections={filtervalue} //To be kept as Select
+              isOpen={isFilterOpen}
+              aria-labelledby="filter"
+             
+            >
+            
+              {
+              webPropertiesKeys.map((x,y) =>
```
Comment (akhilmhdh) [inline packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx:178]: Remove the unshift at top and just put the component directly here and change the name from x and y to better variable names.

File: packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx [hunk 4]
```diff
@@ -131,12 +143,48 @@ export const WebPropertyDetailPage = (): JSX.Element => {
...
+          <SplitItem isFilled />
+          <SplitItem isFilled />
+          <SplitItem>
+            <Select
+              variant={SelectVariant.single}
+              aria-label="filter Input"
+              value="Select Enviroment"
+              onToggle={setIsFilterOpen.toggle}
+              onSelect={(e, value) => {
+                if(value === "Select Enviroment")
```
Comment (akhilmhdh) [inline packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx:164]: put curly

File: packages/manager/package.json [hunk 5]
```diff
@@ -42,14 +42,14 @@
     "@formkit/auto-animate": "1.0.0-beta.5",
     "@hookform/resolvers": "2.9.10",
     "@patternfly/react-charts": "6.94.12",
-    "@patternfly/react-core": "4.264.0",
+    "@patternfly/react-core": "^4.264.0",
     "@patternfly/react-icons": "4.93.0",
```
Comment (akhilmhdh) [inline packages/manager/package.json:46]: Bump up the minor version number

**Review by akhilmhdh — CHANGES_REQUESTED** [review]
Run eslint  command and fix all lint errors

File: packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx (hunk 3 above)
Comment (kotwani2883) [reply packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx:178]: The environment names are coming from the API , the select environment is the default and is not coming in the response

Comment (akhilmhdh) [reply packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx:178]: Yah. So just put it as component here. then do the map of rest of the data coming from api

File: packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx [hunk 6]
```diff
@@ -131,12 +146,63 @@ export const WebPropertyDetailPage = (): JSX.Element => {
...
+                          setFilterByEnv(value as string);
+                        }
+                        setIsFilterOpen.off();
+                      }}
+                      // selections={filtervalue} //To be kept as Select
+                      isOpen={isFilterOpen}
+                      aria-labelledby="filter"
+                    >
+                      {webPropertiesKeys.map((envName) => (
+                        <SelectOption key={uuid()} value={envName} />
```
Comment (akhilmhdh) [inline packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx:179]: may i know the purpose of this UUID generation. 

For unique key may be use a combo of index + env name. That will be unique. And kindly remove this package if not used

Comment (akhilmhdh) [reply packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx:179]: And i mean before the map function.

Apply 

```js
<Select value="Select Environment" />
{map-it}
```

**Review by akhilmhdh — CHANGES_REQUESTED** [review]

File: packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx (hunk 6 above)
Comment (akhilmhdh) [reply packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx:179]: Also itt should be 2._.8 right. Why .9

Comment (kotwani2883) [reply packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx:179]: > may i know the purpose of this UUID generation.
> 
> For unique key may be use a combo of index + env name. That will be unique. And kindly remove this package if not used

Its not a good practice to use index in the key according to eslint

Comment (akhilmhdh) [reply packages/manager/src/views/WebPropertyDetailPage/WebPropertyDetailPage.tsx:179]: Yes it's not. But that's when you use just index only. That's why suggested env-name + index combo. So even if env name ends up duplicate index makes it unique.

It's not good to generrate a uuid on each render.

**Review by riginoommen — APPROVED** [review]

File: packages/cli/package.json [hunk 7]
```diff
@@ -47,8 +47,7 @@
     "ora": "^5.0.0",
     "pretty-bytes": "^5.3.0",
     "read-package-json": "^4.1.1",
-    "read-pkg-up": "^7.0.0",
-    "uuid": "^8.3.2"
+    "read-pkg-up": "^7.0.0"
```
Comment (akhilmhdh) [inline packages/cli/package.json:50]: Hey @kotwani2883 is this by accident

**Review by hybridx — CHANGES_REQUESTED** [review]

File: packages/cli/package.json (hunk 7 above)
Comment (kotwani2883) [reply packages/cli/package.json:50]: Yeah my  bad

**Review by akhilmhdh — APPROVED** [review]

**Comment by akhilmhdh** [comment]
Good job @kotwani2883

**Review by hybridx — APPROVED** [review]

**Pull request merged**